src/main/java/org/jboss/pnc/cleaner/builds/FailedBuildsCleaner.java

```diff
@@ -265,41 +262,43 @@ boolean shouldClean(String groupName, FailedBuildsCleanerSession session) throws
      * @param buildContentId id of the wanted build
      * @return found build record or null
      */
-    private BuildRecordRest getBuildRecord(String buildContentId) throws CleanerException {
+    private Build getBuildRecord(String buildContentId) throws CleanerException {
         logger.debug("Looking for build record with query \"buildContentId==" + buildContentId + "\"");
-        BuildRecordPage page = buildService.getAll(0, 2, null, "buildContentId==" + buildContentId, null, null);
-        if (page != null && page.getContent() != null && page.getContent().size() > 1) {
-            logger.error("Multiple build records found for buildContentId = {}", buildContentId);
-        } else if (page == null || page.getContent() == null || page.getContent().size() == 0) {
-            logger.warn("Build record NOT found for buildContentId = {}", buildContentId);
-            Matcher matcher = buildNumPattern.matcher(buildContentId);
-            if (matcher.matches()) {
-                Integer id = Integer.valueOf(matcher.group(1));
-                logger.debug("Attempting to find build record by id {}", id);
-                BuildRecordSingleton singleton = null;
-                try {
-                    singleton = buildService.getSpecific(id);
-                } catch (WebApplicationException ex) {
-                    int status = ex.getResponse().getStatus();
-                    if (status == Response.Status.NOT_FOUND.getStatusCode()) {
+
+        Matcher matcher = buildNumPattern.matcher(buildContentId);
+        String id = matcher.group(1);
```

Comment:
And also this was supposed to be done after a build cannot be found by buildContentId. I believe we still have some edge cases when buildContentId is left empty, so search by parsed ID is a fallback in case it can be parsed. This should be moved to line 282.

Reply:
OK, I missed that. Moved back